### PR TITLE
Add declare model symbol methods to SymbolManager and Model

### PR DESCRIPTION
**Pull request by ajreynol**
This is in preparation for the symbol manager determining which symbols are printed in the model.

### CODE REVIEW

File: src/expr/symbol_manager.cpp [hunk 1]
```diff
@@ -150,6 +166,36 @@ SymbolManager::Implementation::getExpressionNames(bool areAssertions) const
...
+       ++it)
+  {
+    declareSorts.push_back(*it);
+  }
+  for (TermList::const_iterator it = d_declareTerms.begin();
+       it != d_declareTerms.end();
+       ++it)
+  {
+    declareTerms.push_back(*it);
+  }
```
Comment (nafur) [inline src/expr/symbol_manager.cpp:184]:
```suggestion
  declareSorts.insert(declareSorts.end(), d_declareSorts.begin(), d_declareSorts.end());
  declareTerms.insert(declareTerms.end(), d_declareTerms.begin(), d_declareTerms.end());
```

**Review by nafur — CHANGES_REQUESTED** [review]
One request. A general question: why not return sorts and terms separately in the SymbolManager as well, just like in the Model?

**Comment by ajreynol** [comment]
I've split the access functions in symbol manager for sorts and terms.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**